calyx/src/analysis/control_ports.rs

```diff
@@ -14,132 +14,144 @@ type InvokeMap = HashMap<ir::Id, Vec<Binding>>;
 /// Contains a mapping from name of [ir::CombGroup] to the ports read by the control program
 /// as well as the mapping from invoke statements to the port mappings.
 /// The vector of ports is guaranteed to only contain unique ports.
-pub struct ControlPorts {
+pub struct ControlPorts<const INVOKE_MAP: bool> {
```

Comment:
I feel like @EclecticGriffin will enjoy this type hackery. I really like how this pattern can allow us to build analysis that “only” have code paths that a particular pass cares about.

Reply:
Love const generics!

Reply:
I guess the main difference being that it's a statically known option which is different than the normal type

Reply:
It is nice that the monomorphized version of the code will optimize into the two separate paths!